### PR TITLE
APPPOCTOOL-27 Remove tenant matching validation

### PR DESCRIPTION
**Pull request by pfilippov-epam**
### Purpose

Removes tenant matching validation (token and header value) from request that has empty tenantPermissions.

### Approach
- Includes module's path prefix to the permitted endpoint
- Remove tenant matching validation
- Adjust unit tests

### TODOs and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.

### Learning
_Describe the research stage. Add links to blog posts, patterns, libraries or addons used to solve this problem._

### CODE REVIEW

**Review by dmtkachenko — APPROVED** [review]

File: folio-security/src/main/java/org/folio/security/domain/model/AuthUserPrincipal.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  public String getName() {
+    return userId != null ? userId.toString() : null;
+  }
+
+  /**
+   * Sets userId for {@link AuthUserPrincipal} and returns {@link AuthUserPrincipal}.
+   *
+   * @return this {@link AuthUserPrincipal} with new userId value
+   */
+  public AuthUserPrincipal userId(UUID userId) {
```
Comment (mykolam-epm) [inline folio-security/src/main/java/org/folio/security/domain/model/AuthUserPrincipal.java:24]: Why do this manually and not use Lombok?

Comment (pfilippov-epam) [reply folio-security/src/main/java/org/folio/security/domain/model/AuthUserPrincipal.java:24]: Lombok does not have support for with and setters simultaneously

**Review by OleksiiKuzminov — APPROVED** [review]

File: folio-security/src/test/resources/log4j2.properties [hunk 2]
```diff
@@ -6,6 +6,10 @@ appender.console.type=Console
 appender.console.name=STDOUT
 appender.console.layout.type=PatternLayout
 appender.console.layout.pattern=%d{HH:mm:ss} %-5p %-20.20C{1} %m%n
-rootLogger.level=info
+
+# To reliably capture output from Log4j2 (org.springframework.boot.test.system.OutputCaptureExtension.class)
+appender.console.follow = true
+
+rootLogger.level=debug
```
Comment (mykolam-epm) [inline folio-security/src/test/resources/log4j2.properties:13]: Do we want to have debug the default level?

Comment (pfilippov-epam) [reply folio-security/src/test/resources/log4j2.properties:13]: for tests - yes, to be able to validate debug log level logs

**Review by mykolam-epm — APPROVED** [review]

**Pull request merged**